packages/docs/src/routes/qwikcity/api/index.mdx

```diff
@@ -75,9 +75,9 @@ Use the [`useLocation()`](/qwikcity/api/index.mdx#uselocation) function to retri
 ```tsx
 export interface RouteLocation {
   readonly params: RouteParams; // Record<string,string>
-  readonly href: string;
-  readonly pathname: string;
-  readonly query: Record<string, string>;
+  readonly href: string; // @deprecated Use `url` instead
```

Comment:
can we use the current typing here which is:
```
export interface RouteLocation {
  readonly params: Readonly<Record<string, string>>;
  readonly url: URL;
  /**
   * @deprecated Please use `url` instead of href
   */
  readonly href: string;
  /**
   * @deprecated Please use `url` instead of pathname
   */
  readonly pathname: string;
  /**
   * @deprecated Please use `url` instead of query
   */
  readonly query: URLSearchParams;
  readonly isNavigating: boolean;
}
```
but w/o the deprecated keys 😄

Reply:
> but w/o the deprecated keys 😄

How? 😄 

1:

```ts
export interface RouteLocation {
  readonly params: Readonly<Record<string, string>>;
  readonly url: URL;
  /**
   * @deprecated Please use `url` instead of href
   */
  readonly href: string;
  /**
   * @deprecated Please use `url` instead of pathname
   */
  readonly pathname: string;
  /**
   * @deprecated Please use `url` instead of query
   */
  readonly query: URLSearchParams;
  readonly isNavigating: boolean;
}
```

2:

```ts
export interface RouteLocation {
  readonly params: Readonly<Record<string, string>>;
  readonly url: URL;
  /**
   * Please use `url` instead of href
   */
  readonly href: string;
  /**
   * Please use `url` instead of pathname
   */
  readonly pathname: string;
  /**
   * Please use `url` instead of query
   */
  readonly query: URLSearchParams;
  readonly isNavigating: boolean;
}
```

or 3:

```ts
export interface RouteLocation {
  readonly params: Readonly<Record<string, string>>;
  readonly url: URL;
  readonly isNavigating: boolean;
}
```

or maybe something like this (4):

```ts
export interface RouteLocation {
  readonly params: Readonly<Record<string, string>>;
  readonly url: URL;
  readonly isNavigating: boolean;

  // deprecated...
  readonly href: string;
  readonly pathname: string;
  readonly query: URLSearchParams;
}
```